src/server/json_family.cc

```diff
@@ -326,8 +420,80 @@ OpResult<string> OpDoubleArithmetic(const OpArgs& op_args, string_view key, stri
   return output.as_string();
 }
 
+OpResult<long> OpDel(const OpArgs& op_args, string_view key, string_view path) {
+  long total_deletions = 0;
+  if (path.empty()) {
+    auto& db_slice = op_args.shard->db_slice();
+    auto fres = db_slice.FindExt(op_args.db_ind, key);
+    if (IsValid(fres.first)) {
+      total_deletions += long(db_slice.Del(op_args.db_ind, fres.first));
+    }
+
+    return total_deletions;
+  }
+
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return total_deletions;
+  }
+
+  vector<string> deletion_items;
+  auto cb = [&](const string& path, json& val) { deletion_items.emplace_back(path); };
+
+  json j = result.value();
+  error_code ec = JsonReplace(j, path, cb);
+  if (ec) {
+    VLOG(1) << "Failed to evaulate expression on json with error: " << ec.message();
+    return total_deletions;
+  }
+
+  if (deletion_items.empty()) {
+    return total_deletions;
+  }
+
+  json patch(json_array_arg, {});
+  reverse(deletion_items.begin(), deletion_items.end());  // deletion should finish at root keys.
+  for (const auto& it : deletion_items) {
+    string pointer = ConvertToJsonPointer(it);
+    if (pointer.empty()) {
+      VLOG(1) << "Failed to convert the following JSONPath to pointer: " << it;
+      continue;
+    }
+    total_deletions++;
+    json patch_item(json_object_arg, {{"op", "remove"}, {"path", pointer}});
+    patch.emplace_back(patch_item);
+  }
+
+  jsonpatch::apply_patch(j, patch, ec);
+  if (ec) {
+    VLOG(1) << "Failed to apply patch on json with error: " << ec.message();
+    return 0;
+  }
+
+  SetString(op_args, key, j.as_string());
```

Comment:
so I noticed that `SetString` has `PostUpdate` call but every write operation that mutates an existing value must also have PreUpdate call. Check out `OpRem` function in set_family.cc for example (there are dozens of other examples as well). Please fix "OpDel" accordingly and also other write functions as well.